libopenage/curve/tests/curve_types.cpp

```diff
@@ -386,6 +387,86 @@ void curve_types() {
 		TESTEQUALS(c.get(1), 0);
 		TESTEQUALS(c.get(5), 0);
 	}
+
+	{ // array
+		auto f = std::make_shared<event::EventLoop>();
+
+		Array<int, 4> a(f,0);
+		a.set_insert(time::time_t(1), 0, 0);
+		a.set_insert(time::time_t(1), 1, 1);
+		a.set_insert(time::time_t(1), 2, 2);
+		a.set_insert(time::time_t(1), 3, 3);
+		//a = [[0:0, 1:0],[0:0, 1:1],[0:0, 1:2],[0:0, 1:3]]
+
+		auto res = a.get_all(time::time_t(1));
+		TESTEQUALS(res[0], 0);
+		TESTEQUALS(res[1], 1);
+		TESTEQUALS(res[2], 2);
+		TESTEQUALS(res[3], 3);
+
+		Array<int, 4> other(f,0);
+		other[0].last(999);
+		other[1].last(999);
+		other[2].last(999);
+		other[3].last(999);
+		other.set_insert(time::time_t(1), 0, 4);
+		other.set_insert(time::time_t(1), 1, 5);
+		other.set_insert(time::time_t(1), 2, 6);
+		other.set_insert(time::time_t(1), 3, 7);
+		//other = [[0:999, 1:4],[0:999, 1:5],[0:999, 1:6],[0:999, 1:7]]
+
+
+		a.sync(other, time::time_t(1));
+		//a = [[0:0, 1:4],[0:0, 1:5],[0:0, 1:6],[0:0, 1:7]]
+
+		res = a.get_all(time::time_t(0));
+		TESTEQUALS(res[0], 0);
+		TESTEQUALS(res[1], 0);
+		TESTEQUALS(res[2], 0);
+		TESTEQUALS(res[3], 0);
+		res = a.get_all(time::time_t(1));
+		TESTEQUALS(res[0], 4);
+		TESTEQUALS(res[1], 5);
+		TESTEQUALS(res[2], 6);
+		TESTEQUALS(res[3], 7);
+
+		// Additional tests
+		a.set_insert(time::time_t(2), 0, 15);
+		a.set_insert(time::time_t(2), 0, 20);
+		a.set_replace(time::time_t(2), 0, 25);
+		TESTEQUALS(a.get(time::time_t(2), 0), 25);
+		// a = [[0:0, 1:4, 2:25],[0:0, 1:5],[0:0, 1:6],[0:0, 1:7]]
+
+		a.set_insert(time::time_t(3), 0, 30);
+		a.set_insert(time::time_t(4), 0, 40);
+		a.set_last(time::time_t(3), 0, 35);
+		TESTEQUALS(a.get(time::time_t(3), 0), 35);
```

Comment:
```suggestion
		TESTEQUALS(a.get(4, 0), 35);
```

Changing it like this would check if the keyframe `(t=4, v=40)` got deleted.